plasmapy/constants/atomic.py

```diff
@@ -746,9 +782,56 @@ def ion_mass(argument, Z=1, mass_numb=None):
     <Quantity 9.288122788133088e-26 kg>
     >>> ion_mass('Fe-56')
     <Quantity 9.288122788133088e-26 kg>
+    >>> ion_mass(9.11e-31*u.kg)
+    <Constant name='Electron mass' value=9.10938291e-31 uncertainty=4e-38
+    unit='kg' reference='CODATA 2010'>
+    >>> ion_mass(1.67e-27*u.kg)
+    <Quantity 1.67e-27 kg>
 
     """
 
+    if isinstance(argument, u.Quantity) and Z is None and mass_numb is None:
+
+        try:
+            m_i = argument.to(u.kg)
+        except Exception:
+            raise u.UnitConversionError("If the ion in given as a Quantity, "
+                                        "then it must have units of mass.")
+
+        if np.isclose(m_i.value, const.m_e.value, atol=1e-33):  # positrons
+            return const.m_e
+        elif 1.66e-27 <= m_i.value < 7e-25:  # mass range of known isotopes
+            return m_i
+        else:
+            raise UserWarning("The mass that was inputted to ion_mass and is "
+                              "being returned from ion_mass is outside of the "
+                              "range of known isotopes or electrons/ions.")
+
+    if isinstance(argument, str) and \
+            str(argument).lower() in ['e+', 'positron']:
+        return const.m_e
+
+    if isinstance(argument, str) and \
+            str(argument).lower() in ['e+', 'positron']:
+        return const.m_e
```

Comment:
Another good catch.  A `for` loop would be a much more pythonic way to duplicate lines:

```Python
    for i in range(0,2):
        if isinstance(argument, str) and \
                         str(argument).lower() in ['e+', 'positron']:
            return const.m_e
```

(Just kidding!  I'll delete the duplication.)